### ISSUE TITLE
admin: product collect info - datepicker not localised only YYY-MM-DD

### ISSUE THREAD

**Issue by torvista** [zencart/zencart#3020]
In Admin, edit a product for Date Available.
The datepicker displays correctly and inserts the date using hyphens but this date is not displayed on the preview page and is not stored.

I found this while attempting to implement it for the specials dates. There it is compounded by the code expecting forward slashes as separators and is then processed by zen_date_raw.

I get dizzy looking at the date variables in the language files around this
define('DATE_FORMAT_DATE_PICKER', 'yy-mm-dd');  //Use only 'dd', 'mm' and 'yy' here in any order
which datepicker uses for its display format and hence the post format,
and wonder if some of them should be moved out of harms way/not be part of localisation.

**Comment by mc12345678** [comment]
Is the date being entered a future date, the date of today or a past date?

I note that if the date entered is not in the future then there is no visual indication carried forward that a date was entered.  Going back to the entry page the entered date remains.  Testing was performed on a system that is default installed without localization changes to support date formats in other locations where the neighboring text indicates to enter the date as 'YYYY-MM-DD' which is an ISO 8601 format.

**Comment by torvista** [comment]
Past Date: value is accepted. Not shown on preview page (correctly as per code).
On return to product edit, past date is not shown in field.

Future Date: value is accepted. Is shown on preview page.
On return to product edit, past date is shown in field.

I probably clicked on a past date.

**Comment by torvista** [comment]
in english.php:
```
define('DATE_FORMAT_SHORT', '%m/%d/%Y');  // this is used for strftime()
define('DATE_FORMAT_LONG', '%A %d %B, %Y'); // this is used for strftime()
define('DATE_FORMAT', 'm/d/Y'); // this is used for date()
define('PHP_DATE_TIME_FORMAT', 'm/d/Y H:i:s'); // this is used for date()
define('DATE_TIME_FORMAT', DATE_FORMAT_SHORT . ' %H:%M:%S');
// for now both defines are needed until Spiffy is completely removed.
define('DATE_FORMAT_SPIFFYCAL', 'MM/dd/yyyy');  //Use only 'dd', 'MM' and 'yyyy' here in any order
define('DATE_FORMAT_DATE_PICKER', 'yy-mm-dd');  //Use only 'dd', 'mm' and 'yy' here in any order
```
Note that by default, all are mm/dd/yyyy apart from datepicker.

Changing the datepicker order does change the format that the datepicker places in the field, but this is the only format that the field can handle. Other formats consistent with the localisation do not work/are not saved.

Datepicker uses this constant:
```
  $.datepicker.setDefaults($.extend({}, $.datepicker.regional["<?php echo $_SESSION['languages_code'] == 'en' ? '' : $_SESSION['languages_code']; ?>"], {
      dateFormat: '<?php echo DATE_FORMAT_DATE_PICKER; ?>',
```
So on each page that uses this, I think there needs to be 
a) an echo of the constant as a format hint (in product.php it is hard-coded as YYYY-MM-DD).
b) some intermediate conversion of the datepicker/localised format to the standard format for db insertion/a function to do it.

**Comment by scottcwilson** [comment]
Could we close this ticket and open a new one with a less dramatic title (e.g. Datepicker should hint that required format is sortable date YYYY-MM-DD or something like that)?

**Comment by torvista** [comment]
1) I note the date must be in the future to be shown in the preview.
2) The date format is shown hard-coded as YYYY-MM-DD which is the format the datepicker creates from the calender. So, it is correct, simply not localised in agreement with the language file format settings.

**Comment by scottcwilson** [comment]
Re: 1 - a Date Available in the past doesn't really make sense.  This field is for managing upcoming products.  This is easily changed if you think it should be different - just modify update_product.php and get rid of the tertiary statement for this field and submit a PR. 

On #2 - You are right about the text string YYYY-MM-DD, which is misleading.   You are also right that localization doesn't work as promised.  These issues are both fixed in #3242. 

I did notice another bug in testing this (autocomplete messes up the widget), which is fixed in #3240.

**Comment by scottcwilson** [comment]
Tracking these issues in #3243 so we don't have to keep updating a closed ticket. :)
